### ISSUE TITLE
which performance is better when use crf loss or softmax loss?

### ISSUE THREAD

**Issue by OYE93** [kyzhouhzau/BERT-NER#65]
Hello, thanks for your job. 
I have a question about different loss function, is there any difference for performance when using different loss function, like crf_loss and softmax_loss?
thanks.

**Comment by OYE93** [comment]
I tried the code for crf_loss and softmax_loss, both using default param, the following are the results:
**--crf=False **  
![Selection_041](https://user-images.githubusercontent.com/19822397/62507732-dd7a4280-b837-11e9-895c-9ae34fa6130b.png)
**--crf=True **  
![Selection_040](https://user-images.githubusercontent.com/19822397/62507742-e3702380-b837-11e9-8287-73346682aa38.png)

**Comment by zwd13122889** [comment]
Hey, i want to know the data set.The first raw is word ,the fourth raw is the label, what's the second and third raw meaning?
Another question is the output label_test.txt , its second and third raw are same, does it have another meaning ?

**Comment by OYE93** [comment]
Hi, I think you mean the col, I guess the 2nd col is part-of-speech(POS), the 3rd col is word segmentation.
in label_test.txt, the third col should be the predicted tags, so the 2nd and 3rd col can not the the same. you can compare the 2nd and the 3rd col to evaluate the prediction

**Comment by zwd13122889** [comment]
Thank you, I got it. Another question is that if my dataset don't have the 2nd col (POS) and the 3rd col (word segmentation) ,can this model run?Because in the BERT_NER.py ,i don't see any treatment about the 2nd column and the 3rd.

**Comment by OYE93** [comment]
off course, only the 1st and 4th col are necessary, you just transform your data to 2 col format, you can use the code for training and testing

**Comment by zwd13122889** [comment]
Thank you very much!!!

**Comment by OYE93** [comment]
:)

**Comment by zwd13122889** [comment]
Excuse me, i have another question. Where does the  label_test.txt come from? Man made or machine generated?

**Comment by zwd13122889** [comment]
![微信截图_20191021215821](https://user-images.githubusercontent.com/49578851/67212794-74c25f80-f44f-11e9-8795-bf910f76d2cf.png)
Did this program run successfully?

**Comment by OYE93** [comment]
label_test.txt is generated, seems successful. now you can run this on your own dataset.

**Comment by zwd13122889** [comment]
OK. I run my own data. But i have some problem show in the picture:
![微信截图_20191030151556](https://user-images.githubusercontent.com/49578851/67837507-2afe0700-fb2a-11e9-87c1-60b51d89c4f8.png)
the left is author's data ,the right is mine

**Comment by zwd13122889** [comment]
![微信截图_20191030151556](https://user-images.githubusercontent.com/49578851/67837678-91832500-fb2a-11e9-9a71-2562523ff81b.png)

**Comment by gungor2** [comment]
I just found out crf=False does not work. You always use crf layer.

I raised an issue: 

https://github.com/kyzhouhzau/BERT-NER/issues/88
